bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/listeners/AssessmentMarkerViewDoubleClickListener.java

```diff
@@ -0,0 +1,72 @@
+package edu.kit.kastel.eclipse.grading.view.listeners;
+
+import java.util.NoSuchElementException;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jface.viewers.DoubleClickEvent;
+import org.eclipse.jface.viewers.IDoubleClickListener;
+import org.eclipse.jface.viewers.TreeSelection;
+import org.eclipse.ui.views.markers.MarkerItem;
+
+import edu.kit.kastel.eclipse.common.view.utilities.AssessmentUtilities;
+import edu.kit.kastel.eclipse.grading.view.activator.Activator;
+import edu.kit.kastel.eclipse.grading.view.assessment.ArtemisGradingView;
+import edu.kit.kastel.eclipse.grading.view.assessment.CustomButtonDialog;
+import edu.kit.kastel.sdq.eclipse.grading.api.model.IAnnotation;
+
+public class AssessmentMarkerViewDoubleClickListener implements IDoubleClickListener {
+
+	private ArtemisGradingView gradingView;
+
+	public AssessmentMarkerViewDoubleClickListener(ArtemisGradingView gradingView) {
+		this.gradingView = gradingView;
+	}
+
+	@Override
+	public void doubleClick(DoubleClickEvent event) {
+		if (event.getSelection().isEmpty()) {
+			// No element clicked
+			return;
+		}
+
+		// instanceof is required... :(
+		if (event.getSelection() instanceof TreeSelection) {
+			TreeSelection selection = (TreeSelection) event.getSelection();
+			if (selection.getFirstElement() instanceof MarkerItem) {
+				MarkerItem item = (MarkerItem) selection.getFirstElement();
```

Comment:
This could be simplified using the new instanceof if-matching in java 17 :)

Reply:
Are we using java 17? I think I had issues with even running `mvn package` using maven with Java 17. 
Could also be a problem with my current setup. 🤔

Reply:
mmh maybe a newer eclipse is needed. We are using Eclipse 2021-12 now for building :)

Reply:
Turned out I just never merged your update to java 17 into my branch :D

Reply:
:D ok